### PR TITLE
feat(new sink): New `humio_metrics` sink.

### LINKED ISSUE #3271: New `humio_metrics` sink
Humio can ingest metrics data, as described in the following pages:

* https://docs.humio.com/integrations/data-shippers/beats/metricbeat/
* https://docs.humio.com/integrations/data-shippers/statsd/

Given that Statsd is a more open protocol, **we'd like to _wrap_ the `statsd` sink for the `humio_metrics` sink**. Notice how we [wrap the `splunk_hec` sink for the `humio_logs` sink](https://github.com/timberio/vector/blob/master/src/sinks/humio_logs.rs).

### PR DESCRIPTION
**Pull request by FungusHumungus**
Closes #3271 

This sink sends metric data to [Humio](https://www.humio.com/).

It is a simple wrapper over the `metric_to_log` transform and the `humio_logs` sink.

### CODE REVIEW

File: docs/reference/components/sinks/humio_metrics.cue
Comment (binarylogic) [inline docs/reference/components/sinks/humio_metrics.cue:67]: If these are exactly the same as `humio_logs` you could just do:

```
configuration: components.sinks.humio_logs.configuration
```

Comment (binarylogic) [reply docs/reference/components/sinks/humio_metrics.cue:67]: And the same goes for most of the other settings here as well.

Comment (FungusHumungus) [reply docs/reference/components/sinks/humio_metrics.cue:67]: Oh nice!

That will have to wait until #4359 gets merged.

**Comment by jszwedko** [comment]
Are the advantages/disadvantages of having a separate sink for this vs. just having a `humio` sink that can handle logs and metric data?

**Comment by binarylogic** [comment]
@jszwedko it's mostly about the precedence we've set:

1. Typically the sinks will interface with different endpoints, etc.
2. If we named the sink just `humio` I wonder if it would be obvious to users that they could send metrics to it?

This is a unique case though, in that we are just converting metrics to logs. I would be open to performing this conversion automatically when a `metrics` output is connected to a `logs` input.

**Comment by lukesteensen** [comment]
> I would be open to doing this conversion automatically when a `metrics` output is connected to a `logs` input.

This makes me nervous. We have the whole type checking system set up to catch config errors like this, which are probably more common than wanting an implicit conversion.

**Comment by binarylogic** [comment]
cc @Hoverbear since this is basically doing what's outlined in #4427

File: docs/reference/components/sinks/humio_metrics.cue
Comment (jamtur01) [reply docs/reference/components/sinks/humio_metrics.cue:67]: @FungusHumungus Just an FYI I merged #4359

**Review by jszwedko — APPROVED** [review]
This looks good!

**Review by Hoverbear — APPROVED** [review]

**Comment by binarylogic** [comment]
Nice work! Merging since I've fixed the docs.

**Comment by FungusHumungus** [comment]
Sweet, thanks! I was just about to do it, but I think you did it better than I was planning to.

**Pull request merged**
